cmd/start_server/start_server.go

```diff
@@ -1,23 +1,95 @@
 package main
 
 import (
+	"flag"
 	"fmt"
 	"os"
+	"path"
+	"strconv"
 
 	lsp "github.com/CircleCI-Public/circleci-yaml-language-server/pkg/server"
 )
 
 func main() {
-	if len(os.Args) > 1 {
-		switch os.Args[1] {
-		case "--version":
-			fmt.Println(lsp.GetServerVersion())
+	hostRef := flag.String("host", "", "Hostname of the server")
+	portRef := flag.Int("port", -1, "port number")
+	schemaRef := flag.String("schema", "", "Location of the schema")
+	versionRef := flag.Bool("version", false, "display version")
+	flag.Parse()
+
+	// Parameter: version
+	version := *versionRef
+	if version {
+		fmt.Println(lsp.GetServerVersion())
+		return
+	}
+
+	// Parameter: schema
+	schema := *schemaRef
+	if schema == "" {
+		schema = os.Getenv("SCHEMA_LOCATION")
+
+		if schema == "" {
+			fmt.Print("No schema defined")
+			return
+		}
+
+		if !path.IsAbs(schema) {
+			cwd, err := os.Getwd()
+
+			if err != nil {
+				fmt.Printf("Error while resolving schema path \"%s\"", schema)
+				panic(err)
+			}
+			schema = path.Join(cwd, schema)
+		}
+	}
+
+	// Command: stdio
+	if len(os.Args) > 1 && os.Args[1] == "stdio" {
+		lsp.StartServerStdio(schema)
+		return
+	}
+
+	// Parameter: host
+	host := *hostRef
+	if host == "" {
+		host = os.Getenv("LSP_SERVER_HOST")
+
+		if host == "" {
+			host = "localhost"
+		}
+	}
+
+	// Parameter: port
+	port := *portRef
+	if port == -1 {
+		portEnv := os.Getenv("PORT")
+		if portEnv == "" {
+			fmt.Print("No port provided. Use --port or define PORT variable")
+			return
+		}
+
+		var err error
```

Comment:
I don't think it's necessary to define the err as a var before the next line (use ":=")

Reply:
Actually, this was necessary ^^